.github/workflows/gh-ci.yml

```diff
@@ -1,10 +1,25 @@
+# Auto-generated file. Do not edit manually!
```

Comment:
Using `matrix` for each suffix looks like a good idea as it prevents code duplication.

Reply:
@adamxchen:

I started off with modifying the `.yml` file but it became unwieldy with the amount of flows I wanted to have, coupled with the fact that [GH's flavor of YAML doesn't support anchors](https://github.com/actions/runner/issues/1182). Even if anchors were supported, my understanding is that anchors aren't expressive enough to really eliminate a lot of the duplication...

The generation via Gradle that I'm proposing here is pretty primitive and I'm sure there are ways to improve it. I haven't looked at splitting into multiple YAML files, and IDK how much benefit it would bring currently since the flows we have so far are nearly identical anyway... but if we have additional types of flows in the future then surely that might become more interesting then. There could also be a better processor than Gradle for spitting out the YAML. I hope there will be continued investment in this area and I wouldn't be vexed if everything in this PR ended up replaced by better things.

That being said, I think there is a benefit in merging this as it speeds up the build.

@sushantmane:

Are there more opportunities to use matrix in other parts of the YAML file? I'm not sure I get what you're saying.

Reply:
@FelixGV  - Currently we have one job for each letter A..Z. If we use a matrix and run a job matching that prefix then we need only one integrationTest block in `gh-ci.yml`

```
IntegrationTestsA:
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest]
        jdk: [11]
        prefix: [A B C D etc]


```

Reply:
Ok I see. That would be better than copy pasting, for sure, but IDK if it would be better the dynamically generating the file.

For now I call the buckets just letters but in a later iteration I would like to give each bucket a more meaningful name, e.g. A_Da_Vinci_1, B_Da_Vinci_2, C_Active_Active_1, D_Active_Active_2, E_Helix, etc... I would like a single source of truth for the bucket names, which is achieved by the current implementation.

Reply:
Well, we could generate a prefix array with the same Gradle script and keep a single source of truth.  

If it requires more work then let's skip it for now. 😄

Reply:
Yes, should be possible.

Reply:
BTW, are we planning to run on other OS and Java versions? Because it seems like the matrix is useless right now... 🤷

Reply:
Originally I wanted to run CI on both macOS & ubuntu but on macOS hosts, it takes longer to finish some tests IIRC. AFA JDK is concerned, kept it in the matrix as a placeholder to be able to run tests on other JDKs as well in the future. 

Please feel free to remove it. Thanks.

Reply:
Ok got it. I'll leave the matrix untouched for now... but maybe we can look into it if/when we refactor the integration tests to be part of the matrix.